### PR TITLE
Add Databricks config to Spark Job

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Add Databricks config to Spark Conf. propeller will read the databricks config, and send a job request to the dbx platform.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Here is an example to run databricks plugin. `databricks_conf` is nested dict, and structured is defined [here](https://docs.databricks.com/dev-tools/api/2.0/jobs.html#request-structure)

```python
from random import random

import flytekit
from operator import add

from flytekit import task, workflow
from flytekitplugins.spark import Spark


@task
def t1(partitions: int) -> int:
    return partitions


databricks_conf = {
    "run_name": "flytekit databricks plugin example",
    "new_cluster": {
        "spark_version": "11.0.x-scala2.12",
        "node_type_id": "r3.xlarge",
        "aws_attributes": {
            "availability": "ON_DEMAND",
            "instance_profile_arn": "arn:aws:iam::590435413560:instance-profile/databricks-s3-role",
        },
        "num_workers": 4,
    },
    "timeout_seconds": 3600,
    "max_retries": 1,
}


@task(task_config=Databricks(databricks_conf=databricks_conf))
def my_spark(partitions: int) -> float:
    n = 1000 * partitions

    def f(_: int) -> float:
        x = random() * 2 - 1
        y = random() * 2 - 1
        return 1 if x ** 2 + y ** 2 <= 1 else 0

    spark = flytekit.current_context().spark_session
    count = spark.sparkContext.parallelize(range(1, n + 1), partitions).map(f).reduce(add)
    pi = 4.0 * count / n
    print("Pi is roughly %f" % pi)
    return pi


@workflow
def my_wf(partitions: int = 5):
    res = t1(partitions=partitions)
    my_spark(partitions=res)


if __name__ == "__main__":
    my_wf()

```

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: plugins/flytekit-spark/flytekitplugins/spark/task.py [hunk 1]
```diff
@@ -27,6 +29,7 @@ class Spark(object):
 
     spark_conf: Optional[Dict[str, str]] = None
     hadoop_conf: Optional[Dict[str, str]] = None
+    databricks_conf: typing.Optional[dict] = None
```
Comment (tmathur-hbo) [inline plugins/flytekit-spark/flytekitplugins/spark/task.py:32]: nit: define the type of dict

**Comment by pingsutw** [comment]
blocked by https://github.com/flyteorg/flyteidl/pull/351

File: flytekit/core/python_auto_container.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import importlib
 import re
+import sys
```
Comment (wild-endeavor) [inline flytekit/core/python_auto_container.py:5]: is this needed?

File: flytekit/bin/entrypoint.py [hunk 3]
```diff
@@ -445,14 +445,14 @@ def _pass_through():
...
     output_prefix,
-    raw_output_data_prefix,
-    test,
-    prev_checkpoint,
-    checkpoint_path,
-    dynamic_addl_distro,
-    dynamic_dest_dir,
-    resolver,
-    resolver_args,
+    raw_output_data_prefix=None,
```
Comment (wild-endeavor) [inline flytekit/bin/entrypoint.py:448]: databricks will use a different entrypoint file uploaded to dbfs right?

**Review by wild-endeavor — COMMENTED** [review]
@kumare3 @EngHabu should we make this a separate task?  It feels just ever so slightly weird to me to see `databricks` as a top level argument to the spark task, even if it is optional.

File: flytekit/bin/entrypoint.py (hunk 3 above)
Comment (pingsutw) [reply flytekit/bin/entrypoint.py]: yes, sorry. I reverted it.

File: flytekit/core/python_auto_container.py (hunk 2 above)
Comment (pingsutw) [reply flytekit/core/python_auto_container.py]: removed

File: flytekit/core/python_auto_container.py [hunk 4]
```diff
@@ -191,6 +192,7 @@ def name(self) -> str:
     def load_task(self, loader_args: List[Union[T, ModuleType]]) -> PythonAutoContainerTask:
         _, task_module, _, task_name, *_ = loader_args
 
+        sys.path.append(".")
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply flytekit/core/python_auto_container.py]: removed it

**Review by wild-endeavor — APPROVED** [review]
